tooling/nargo/src/lib.rs

```diff
@@ -42,9 +42,50 @@ pub fn prepare_dependencies(
     }
 }
 
+// We will pre-populate the file manager with all the files in the package
+// This is so that we can avoid having to read from disk when we are compiling
+//
+// This does not require parsing because we are interested in the files under the src directory
+// it may turn out that we do not need to include some Noir files that we add to the file
+// manager
+pub fn insert_all_files_for_package_into_file_manager(
+    package: &Package,
+    file_manager: &mut FileManager,
+) {
+    // Start off at the root directory of the package and add all of the files located
+    // in that directory.
+    let root_path = package.root_dir.clone();
+
+    // Get all files in the package and add them to the file manager
+    let paths = get_all_paths_in_dir(&root_path).expect("could not get all paths in the package");
```

Comment:
This can get expensive as it's reading in stuff outside of just the `src` dir so if there's other stuff inside the Nargo project such as a `node_modules` directory, we could be sucking up gigabytes of data here.

We should read the entrypoint from the `Nargo.toml` file and just read from that directory.

Reply:
Not saying it needs to be handled right now, just making a note.

Reply:
yes sir, will fixup once the wasm codepath is working

Reply:
Modified here: https://github.com/noir-lang/noir/pull/3760/commits/3e26e3ca4f0820006ed729903a94d34486f993bc